chapters/en/slides/module7/module7_04.Rmd

```diff
@@ -0,0 +1,342 @@
+
+Notes:
+Now that we know how Altair creates charts,
+we can also understand why there is an issue with large datasets.
+
+You can see the JSON chart spec printed in this slide again.
+Note that the data here is just a string value pointing to a URL
+because we used `data.cars.url` 
+to create the chart.
+
+If we instead of a URL would use a pandas dataframe (e.g. `data.cars()`)
+**all** the data would be included in the chart specification.
+In the JSON format,
+this is almost 4500 lines of code!
+There is not enough room to show that in this slide,
+but you can click the link under the code
+to see an example of what it would look like.
```

Comment:
Is 4500 lines of code a problem? Please explain in this transcript why.

```suggestion
Now that we know how Altair creates charts,
we can also understand why there is an issue with large datasets.

You can see the JSON chart spec printed in this slide again.
Note that the data here is just a string value pointing to a URL
because we used `data.cars.url` 
to create the chart.

If we had used a pandas dataframe (e.g. `data.cars()`) to create
the chart (instead of a URL) 
**all** of the data would have been included in the chart specification.

In the JSON format,
this is almost 4500 lines of code!
There is not enough room to show that in this slide,
but you can click the link under the code
to see an example of what it would look like.
```